### PR TITLE
[[ TimeZoneLibrary ]] Add a time zone handling library

### PR DESCRIPTION
**Pull request by livecodeali**
The library contains three handlers:
* `ToUniversalTime(pSeconds, pTimeZone)` - convert from local time in the specified time zone to universal time
* `FromUniversalTime(pSeconds, pTimeZone)` - convert from universal time to local time in the specified time zone 
* `TimeZones()` - list of valid time zones, one per line

At the moment, there are a couple of things done manually here that could be baked into the build process:
- The list of timezones is hard-coded into the LCB source file - ideally we could inject this from a list built when the database files are extracted.
- All the resources should possibly be a prebuilt.

The changes in 3877678 should be applied to the source files whenever they are updated.

### CODE REVIEW

**Comment by montegoulding** [comment]
Hmm... yeah there's a lot of potential binary file churn here so it would be nice as a prebuilt

**Comment by montegoulding** [comment]
Also I think you need to still add to the manifest. TBH the patch is so large chrome is having a very hard time scrolling so maybe it's there and I didn't see it!

**Comment by runrevmark** [comment]
Would it be possible to include the source text versions of the tz files in the repo, and then run the tz compiler at build time to build the binary files? Or even better, have it download the source for the version it is building against at build time and build the tz binary files into the extension?

**Comment by livecodeali** [comment]
I think what you are suggesting is the first point in my list of things that could be baked into the build process. Just need to figure out a cross-platform way to actually build them - I'm sure that could be extracted from the Makefile.

**Comment by livecodeali** [comment]
Build failure on Windows:
```
"C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\livecode.sln" (default target) (1) ->
"C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\LiveCode-all.vcxproj.metaproj" (default target) (2) ->
"C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\extensions.vcxproj.metaproj" (default target) (82) ->
"C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\cross-platform-extension-dependencies.vcxproj.metaproj" (default target) (83) ->
"C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\libraries\timezone\tz\tz.vcxproj.metaproj" (default target) (87) ->
"C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\libraries\timezone\tz\tz-build.vcxproj" (default target) (88) ->
(ClCompile target) -> 
  c:\vulcan\community\worker\try-community-x86-win32\worktree\extensions\libraries\timezone\tz\private.h(158): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory [C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\libraries\timezone\tz\tz-build.vcxproj]
  c:\vulcan\community\worker\try-community-x86-win32\worktree\extensions\libraries\timezone\tz\private.h(158): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory [C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\libraries\timezone\tz\tz-build.vcxproj]
  c:\vulcan\community\worker\try-community-x86-win32\worktree\extensions\libraries\timezone\tz\private.h(158): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory [C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\libraries\timezone\tz\tz-build.vcxproj]
  c:\vulcan\community\worker\try-community-x86-win32\worktree\extensions\libraries\timezone\tz\private.h(158): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory [C:\vulcan\Community\worker\try-community-x86-win32\worktree\build-win-x86\livecode\extensions\libraries\timezone\tz\tz-build.vcxproj]
```

**Comment by livecodeali** [comment]
Presumably we need to build with HAVE_UNISTD_H undef'd on windows.

**Comment by livecodeali** [comment]
This is still not in the ideal form, but I don't think it's a priority to make it a prebuilt / download source at build time and apply patch, so it's probably ready for review. At least there are no binary files.

File: extensions/libraries/timezone/tests/timezone.livecodescript [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+on TestSetup
+	TestLoadExtension "com.livecode.library.timezone"
```
Comment (montegoulding) [inline extensions/libraries/timezone/tests/timezone.livecodescript]: @livecodeali do you need to skip windows here because the resources aren't built?

**Comment by livecodeali** [comment]
@montegoulding Fixed :-)

**Comment by montegoulding** [comment]
@livecode-vulcan review ok c1736c7

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok c1736c7e91d243f27f24a8f6100683b8d0d42f16

**Comment by livecode-vulcan** [comment]
😞 test failure c1736c7e91d243f27f24a8f6100683b8d0d42f16

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1265)
* try-community-armv6-android-api9: [failure](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1281)
* try-community-js-emscripten-sdk1.35: [failure](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1255)
* try-community-universal-ios-iphoneos11.2: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/208)
* try-community-universal-ios-iphonesimulator11.2: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/206)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1259)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1285)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1265)
* try-community-x86-linux-debian8: [failure](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1280)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1264)
* try-community-x86_64-linux-debian8: [failure](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1274)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1316)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1255)

**Comment by livecodeali** [comment]
Just want to make sure this works now
@livecode-vulcan review ok f36ac04

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok f36ac0405560385c12fa4175312e9f5d0d56cafd

**Comment by livecode-vulcan** [comment]
😞 test failure f36ac0405560385c12fa4175312e9f5d0d56cafd

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1272)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1288)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1262)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/215)
* try-community-universal-ios-iphonesimulator11.2: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/213)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1266)
* try-community-universal-mac-macosx10.9: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1292)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1272)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1287)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1271)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1281)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1323)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1262)

**Comment by montegoulding** [comment]
@livecode-vulcan retry f36ac04

**Comment by livecode-vulcan** [comment]
😞 test failure f36ac0405560385c12fa4175312e9f5d0d56cafd

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1278)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1294)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1268)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/221)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/219)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1272)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1298)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1278)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1293)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1277)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1287)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1329)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1268)

**Comment by montegoulding** [comment]
@livecode-vulcan retry f36ac04

**Comment by livecode-vulcan** [comment]
😞 test failure f36ac0405560385c12fa4175312e9f5d0d56cafd

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1280)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1296)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1270)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/223)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/221)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1274)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1300)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1280)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1295)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1279)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1289)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1331)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1270)

**Comment by livecodeali** [comment]
@livecode-vulcan retry f36ac04

**Comment by livecode-vulcan** [comment]
😎 test success f36ac0405560385c12fa4175312e9f5d0d56cafd

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1285)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1301)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1275)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/228)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/226)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1279)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1305)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1285)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1300)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1284)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1294)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1336)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1275)

**Pull request merged**